### PR TITLE
React to Logging API changes

### PR DESCRIPTION
**Pull request by BrennanConroy**

### CODE REVIEW

**Comment by BrennanConroy** [comment]
Meant to open againt rel/2.0.0-preview1

File: src/EFCore.Design/Design/Internal/DatabaseOperations.cs [hunk 1]
```diff
@@ -58,8 +58,8 @@ public class DatabaseOperations
 
             var services = _servicesBuilder.Build(provider);
 
-            var loggerFactory = services.GetService<ILoggerFactory>();
-            loggerFactory.AddProvider(new LoggerProvider(categoryName => new OperationLogger(categoryName, _reporter)));
+            var loggerFactory = services.GetService<ILoggerFactory>() as LoggerFactory;
+            loggerFactory?.AddProvider(new LoggerProvider(categoryName => new OperationLogger(categoryName, _reporter)));
```
Comment (Eilon) [inline src/EFCore.Design/Design/Internal/DatabaseOperations.cs]: Hmm this is suspicious... what if it was a customer ILoggerFactory? What should happen?

Comment (BrennanConroy) [inline src/EFCore.Design/Design/Internal/DatabaseOperations.cs]: We could allow it to call the ILoggerFactory.AddProvider method and just pragma the warning, then remove it in 3.0.0 when the API gets deleted.

**Comment by smitpatel** [comment]
PR in logging repo for changes?

**Comment by BrennanConroy** [comment]
https://github.com/aspnet/Logging/pull/605

File: src/EFCore.Design/Design/Internal/DatabaseOperations.cs (hunk 1 above)
Comment (ajcvickers) [inline src/EFCore.Design/Design/Internal/DatabaseOperations.cs]: What is the underlying change that results in this change being needed?

Comment (BrennanConroy) [inline src/EFCore.Design/Design/Internal/DatabaseOperations.cs]: https://github.com/aspnet/Logging/pull/605/files#diff-528cfd1fcf044b63339b2b84f3b77f84R28

Comment (bricelam) [inline src/EFCore.Design/Design/Internal/DatabaseOperations.cs]: > when the API gets deleted

This functionality seems pretty fundamental to a logging abstraction. What's the story going forward? Is the `ILoggerFactory` interface going away in favor of making `LoggerFactory` a public type in `Microsoft.Extensions.Logging.Abstractions`?

**Comment by ajcvickers** [comment]
@BrennanConroy @Eilon Where can we find the design rationale around these changes and documentation or other information about how the logging abstractions are now intended to be used?

**Comment by Eilon** [comment]
I'll set up a mtg for Monday.

**Comment by BrennanConroy** [comment]
This issue is also a good reference for reasons behind the change https://github.com/aspnet/Logging/issues/420

**Comment by BrennanConroy** [comment]
🆙 📅

**Pull request merged**